### PR TITLE
Plumbs together excise step

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-832

### What?

The excise step is quite similar to the additional code step. It uses
slightly different namespacing for the session and filters on different
additional codes. We may want to consider a refactor separately to DRY up the two steps/reuse behaviour as much as possible.

Excise needs to _always_ pull from the uk data source (like VAT)

This adds:

- [x] The excise step model for validating and persisting the session
  attributes
- [x] The excise controller
- [x] The meaning of an excise measure type in the measure type api model
- [x] The excise show view
- [x] The excise next/previous step paths
- [x] Locale strings for error message validations
- [x] Specs covering the new step and related classes
- [x] Adds confirmation page plumbing

### Why?

I am doing this because:

- I've gone for the full list of supported measure types now since I know I'm going to need them later down the line.
- I've avoided sharing behaviour because the concepts are similar but have separate copy, session keys, etc. If we decide add more measure-filtering additional code steps in future then I definitely think the complexity/DRY battle will be won for sharing behaviour between the steps

### CODE REVIEW

**Comment by willfish** [comment]
image

**Comment by willfish** [comment]
image

**Comment by willfish** [comment]
image

File: app/models/steps/excise.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  class Excise < Steps::Base
+    STEPS_TO_REMOVE_FROM_SESSION = %w[].freeze
+
+    include CommodityHelper
+
+    attribute :measure_type_id, :string
+    attribute :additional_code, :string
+
+    validates :measure_type_id, presence: true
+    validates :additional_code, presence: true
```
Comment (jebw) [inline app/models/steps/excise.rb:11]: Does there need to be additional validation around the codes supplied by the user?

**Review by jebw — COMMENTED** [review]
Looks good but couple of questions around the method on UserSession and wondering if theres any way to validate the additional_codes supplied by the user to the Excise model?

File: app/models/user_session.rb [hunk 2]
```diff
@@ -205,6 +218,10 @@ def additional_codes
     (additional_code_uk.values + additional_code_xi.values).compact.join(', ')
   end
 
+  def excise_additional_codes
```
[the remark this answers is not in the document]
Comment (willfish) [reply app/models/user_session.rb:221]: Sounds good :+1:

File: app/models/steps/excise.rb (hunk 1 above)
Comment (willfish) [reply app/models/steps/excise.rb:11]: Talked this through. :+1:

**Review by jebw — APPROVED** [review]

**Pull request merged**